qutip/core/data/norm.pyx

```diff
@@ -61,6 +63,21 @@ cpdef double trace_csr(CSR matrix, sparse=False, tol=0, maxiter=None) except -1:
         total += math.sqrt(abs(eigs[i]))
     return total
 
+cpdef double trace_dense(Dense matrix) except -1:
+    """Compute the trace norm relaying scipy for dense operations."""
+    return scipy.linalg.norm(matrix.as_ndarray(), 'nuc')
+
+cpdef double trace_csr(CSR matrix, sparse=False, tol=0, maxiter=None) except -1:
+    # For column and row vectors we simply use the l2 norm as it is equivalent
+    # to the trace norm.
+    if matrix.shape[0]==1 or matrix.shape[1]==1:
+        return l2_csr(matrix)
+
+    if sparse:
```

Comment:
There is no way to specify `sparse=True` via the data layer, so I propose we just remove it entirely and combine `_trace_csr` with this method.